### ISSUE TITLE
Corner glitch when using wxCheckBox in place of wxStaticBox label

### ISSUE THREAD

**Issue by charvey2718** [wxWidgets/wxWidgets#22940]
### Description

When placing a wxCheckBox at the top of a wxStaticBox, as per the constructor, [here](https://docs.wxwidgets.org/3.2.0/classwx_static_box.html#aedd6c9352e976dcc55e8112143cca6f7), there is a break in the surrounding box in the top-right corner.

![2](https://user-images.githubusercontent.com/59702518/199719776-ef9b7bf1-a5ee-4c3e-aa63-a7f351196b14.png)

No break in the surrounding box is expected. The break disappears if I replace the wxCheckbox with a wxString label.

### Minimum working example

The issue is reproduced in the following MWE:
![1](https://user-images.githubusercontent.com/59702518/199721959-48fef7e5-bb42-47fc-8adf-33bbfc2d97ca.PNG)
```
#include <wx/wx.h>

class MyFrame:public wxFrame {
public:
    MyFrame(wxWindow* parent);
};

class MyApp:public wxApp {
public:
    bool OnInit() {
        wxFrame* frame=new MyFrame(NULL);
        SetTopWindow(frame);
        frame->Show();
        return true;                    
    }
};

DECLARE_APP(MyApp);
IMPLEMENT_APP(MyApp);

MyFrame::MyFrame(wxWindow* parent):wxFrame(parent,wxID_ANY,_("StatBoxMWE"),wxDefaultPosition,wxDefaultSize,wxDEFAULT_FRAME_STYLE) {
	SetClientSize(wxSize(100,100));
	wxBoxSizer* mainSizer=new wxBoxSizer(wxVERTICAL);
	
	wxCheckBox* cb=new wxCheckBox(this,wxID_ANY,_("checkbox"),wxDefaultPosition,wxDefaultSize,wxALIGN_LEFT);
	wxStaticBox* sb=new wxStaticBox(this,wxID_STATIC,cb);
	wxStaticBoxSizer *sbs=new wxStaticBoxSizer(sb,wxVERTICAL);
	
	wxStaticText* text=new wxStaticText(sb,wxID_ANY,_("hello"),wxDefaultPosition,wxDefaultSize,wxALIGN_LEFT);
	sbs->Add(text,0,wxALL,5);
	
	mainSizer->Add(sbs,1,wxEXPAND|wxALL,10);
	SetSizer(mainSizer);
}
```

### Platform and version information

- wxWidgets version: 3.2.1 (also seen on 3.1.7)
- wxWidgets port: wxMSW
- OS: Windows 10 Enterprise

**Comment by PBfordev** [comment]
As I wrote on the forums, the issue can be reproduced with the unmodified _widgets_ sample:
1. Launch the sample.
2. In the widget list select _Native_ / _Static_.
3. Observe that there is no glitch.
4.  In _Set style_ check _Checkable box_.
5. The glitch appears.

**Comment by vadz** [comment]
Yes, I can see this, thanks. Will try to debug.

**Comment by vadz** [comment]
Copying parts of the static box to `wxPaintDC` used wrong coordinates, fixed now and this will have to be applied to 3.2.

However in master we actually shouldn't use the custom paint code for `wxStaticBox` at all, i.e. do as if `msw.staticbox.optimized-paint` system option were always set to 0 because it's fine to let the control paint itself there because there is no flicker when using `WS_EX_COMPOSITED`. Unfortunately we still need to keep all this code just in case we need to change the label colour, but we can at least avoid using it by default. And maybe we can still get rid of it.

Anyhow, I've done all this in #22952 now, please test it and let me know if you see any new problems with it (the original one should be fixed).

**Comment by charvey2718** [comment]
Many thanks, vadz.
I'm not really familiar with how Github works, but am willing to learn.
What's the correct way to test your commits? Should I download and compile the msw-statbox-paint branch, and test the widgets example locally?

**Comment by vadz** [comment]
Yes, please get the branch (you can use `git fetch https://github.com/vadz/wxWidgets.git && git switch msw-statbox-paint` if your Git is not too ancient) and build the library from it -- and then build either the sample or, even better, your application using this version. TIA!

**Comment by charvey2718** [comment]
The git command you suggested cloned the repo but then said invalid reference for the branch. (I'm on the latest version of git.) Nevermind, I eventually figured out `git clone --recurse-submodules --branch msw-statbox-paint https://github.com/vadz/wxWidgets.git` and it seemed to work okay.

Unfortunately, compiling failed with the error:

```
../src/msw/statbox.cpp: In member function 'virtual bool wxStaticBox::SetForegroundColour(const wxCo
lour&)':
../src/msw/statbox.cpp:231:26: error: expected type-specifier before 'wxStaticText'
         m_labelWin = new wxStaticText(this, wxID_ANY, GetLabel());
                          ^~~~~~~~~~~~
make: *** [Makefile:29170: corelib_msw_statbox.o] Error 1

```
I checked and statbox.cpp didn't include `"#include "wx/stattext.h"`. I added it to line 34 and this seemed to resolve the issue as it then compiled.

I then compiled the widgets sample and my own application, and the staticbox with checkbox now both look fine.

![Capture1](https://user-images.githubusercontent.com/59702518/200704524-f746289e-519e-445a-bc69-5151218ca2b4.PNG)
![Capture2](https://user-images.githubusercontent.com/59702518/200704526-996cc57f-8f16-4d12-872e-abe557ade7db.PNG)


It looks like your commits do indeed solve the problem (the minor issue above notwithstanding).

Thank you!
